### PR TITLE
Creating nvidia repo file and user instructions

### PR DESCRIPTION
**Pull request by chrisdats**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [ ] The toolchain has been rebuilt successfully (or no changes were made to it)
- [ ] The toolchain/worker package manifests are up-to-date
- [ ] Any updated packages successfully build (or no packages were changed)
- [ ] All package sources are available
- [ ] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [ ] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [ ] All source files have up-to-date hashes in the `*.signatures.json` files
- [ ] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [ ] Documentation has been updated to match any changes to the build system
- [ ] Ready to merge

---

###### Summary 
This PR adds documentation to explain how users can configure their package manager to access the NVIDIA PMC RPMs. The documentation also includes the NVIDIA Licensing statement to meet our legal requirements.

###### Change Log  
- Creation of nvidia.md documentation
- Creation of mariner-nvidia.repo file

###### Does this affect the toolchain?  

**YES**
NO

###### Test Methodology

- Pipeline build id: xxxx

### CODE REVIEW

File: toolkit/docs/nvidia/mariner-nvidia.repo [hunk 1]
```diff
@@ -0,0 +1,9 @@
+[mariner-official-nvidia]
```
Comment (PawelWMS) [inline toolkit/docs/nvidia/mariner-nvidia.repo:1]: Why not make this a subpackage for `mariner-repos`?

Comment (PawelWMS) [inline toolkit/docs/nvidia/mariner-nvidia.repo:1]: Please also update the documentation about available .repo files under `toolkit/docs/building/building.md`. Search for "REPO_LIST=...".

File: toolkit/docs/nvidia/nvidia.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+## Overview
+The following documentation describes how to access CBL-Mariner packages from the NVIDIA RPM repository at [packages.microsoft.com](https://packages.microsoft.com/cbl-mariner/1.0/prod/NVIDIA/)
+
+## Licensing
+The software in the NVIDIA RPM repository is subject to the following:
+
+> NVIDIA Software. The software may include components developed and owned by NVIDIA Corporation or its licensors. The use of these components is governed by the NVIDIA end user license agreement located at [https://www.nvidia.com/content/DriverDownload-March2009/licence.php?lang=us](https://www.nvidia.com/content/DriverDownload-March2009/licence.php?lang=us). 
+
+## Instructions
+The following instructions register the nvidia package store with the package manager.
```
Comment (PawelWMS) [inline toolkit/docs/nvidia/nvidia.md:12]: Adding the mentioned subpackage would do these steps for the user, so maybe we won't need this documentation.

**Review by PawelWMS — CHANGES_REQUESTED** [review]
Could you also clean-up the PR summary a bit? Remove unused sections and fill the checkboxes, please.

File: toolkit/docs/nvidia/mariner-nvidia.repo (hunk 1 above)
Comment (jslobodzian) [reply toolkit/docs/nvidia/mariner-nvidia.repo:1]: because we want this to be a manual process by design, as I mentioned last week.

Comment (jslobodzian) [reply toolkit/docs/nvidia/mariner-nvidia.repo:1]: This is not required in this scenario.

File: toolkit/docs/nvidia/nvidia.md (hunk 2 above)
Comment (jslobodzian) [reply toolkit/docs/nvidia/nvidia.md:12]: These manual steps are by design.  As mentioned last week, we are mirroring NVIDIA's licensing agreement and approach.   You must take a manual action to get this.

File: toolkit/docs/nvidia/nvidia.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+```ls
+# Navigate to the package manager configuration file directory
+cd /etc/yum.repos.d
+
+# Copy the configuration to your directory to register the NVIDIA RPM repository with your package manager
+sudo wget https://raw.githubusercontent.com/microsoft/CBL-Mariner/toolkit/docs/nvidia/mariner-nvidia.repo
+```
```
Comment (jslobodzian) [inline toolkit/docs/nvidia/nvidia.md:19]: We might want to add one more instruction here:
sudo dnf install <whatever-the-cuda-package-name-is>

**Review by PawelWMS — APPROVED** [review]

**Review by jslobodzian — APPROVED** [review]

**Pull request merged**